electricitymap/contrib/libs/models/events.py

```diff
@@ -0,0 +1,427 @@
+from abc import ABC, abstractmethod
+from datetime import datetime, timedelta, timezone
+from logging import Logger
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, validator
+
+from electricitymap.contrib.config import EXCHANGES_CONFIG, ZONES_CONFIG, ZoneKey
+from electricitymap.contrib.libs.models.constants import VALID_CURRENCIES
+
+LOWER_DATETIME_BOUND = datetime(2000, 1, 1, tzinfo=timezone.utc)
+
+
+class ProductionMix(BaseModel):
+    """
+    Containts the production mix for a zone at a given time.
+    All values are in MW.
+    """
+
+    biomass: Optional[float] = None
+    coal: Optional[float] = None
+    gas: Optional[float] = None
+    geothermal: Optional[float] = None
+    hydro: Optional[float] = None
+    nuclear: Optional[float] = None
+    oil: Optional[float] = None
+    solar: Optional[float] = None
+    unknown: Optional[float] = None
+    wind: Optional[float] = None
+
+    def set_value(self, mode: str, value: float) -> None:
```

Comment:
It would be nice to have a helper function for it if you have two different kind of say solar in the source data (Thermal and PV) and it would easily sum it up in a tested and documented function.

Reply:
The reason we added this is if someone wants to set them through a loop.
```
for mode, value in production_modes:
   prod_mix.set_value(mode, value)
```
This can't be done directly through the attributes. So we should have a setter here.